TheJumpCloud/support
# Sa 3167 bring your own cert

## Issues
* [SA-3167](https://jumpcloud.atlassian.net/browse/Sa-3167) - SA-3167-Bring-Your-Own-Cert

## What does this solve?
This enables admins to import their own certificates.
## Is there anything particularly tricky?
N/A
## How should this be tested?
1. PFX Import: Create a .pfx file that contains a cert and key. Use the extract instructions in the readme.md file 
* Copy and past the cert.pem and key.pem files to Cert folder, then follow the instruction below. 
2. PEM import: add your cert and key files to the Cert folder. Start radius deployment script, press 2. This should ask you to input your key password that will then saved to ENV. After you input your password, check your env variables if the password is saved to `certKeyPass` variable.
3. Delete and import a different certificate and key with a different password. Press 2, this should tell that the password does not work with the new key, then it will prompt you to enter a password for the new key that will then get saved to ENV.


## Screenshots


[SA-3167]: https://jumpcloud.atlassian.net/browse/SA-3167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
